core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-favorite-page/store/dot-favorite-page.store.ts

```diff
@@ -0,0 +1,138 @@
+                            );
+                        }
+
+                        return this.dotWorkflowActionsFireService.publishContentletAndWaitForIndex<DotCMSContentlet>(
+                            'Screenshot',
+                            {
+                                screenshot: id,
+                                title: formData.title,
+                                url: formData.url,
+                                order: formData.order
+                            },
+                            individualPermissions
+                        );
+                    }),
+                    take(1),
+                    tapResponse(
+                        () => {
+                            this.patchState({ closeDialog: true, loading: false });
+                        },
+                        (error: HttpErrorResponse) => {
+                            this.dotHttpErrorManagerService.handle(error);
+                            return of(null);
+                        }
+                    )
+                );
+            })
+        );
+    });
+
+    constructor(
+        private dotCurrentUser: DotCurrentUserService,
+        private dotRolesService: DotRolesService,
+        private dotMessageService: DotMessageService,
+        private dotHttpErrorManagerService: DotHttpErrorManagerService,
+        private dotTempFileUploadService: DotTempFileUploadService,
+        private dotWorkflowActionsFireService: DotWorkflowActionsFireService
+    ) {
+        super(null);
+    }
+
+    /**
+     * Sets initial state data from props, roles and current logged user data
+     *
+     * @param {DotFavoritePageProps} props
+     * @memberof DotFavoritePageStore
+     */
+    setInitialStateData(props: DotFavoritePageProps): void {
+        const propsData = {
+            isAdmin: props.pageState.user.admin,
+            imgWidth: parseInt(props.pageState.params.viewAs.device?.cssWidth, 10) || 1024,
+            imgHeight:
+                parseInt(props.pageState.params.viewAs.device?.cssHeight, 10) ||
+                (parseInt(props.pageState.params.viewAs.device?.cssWidth, 10) || 1024) /
+                    IMG_RATIO_43
+        };
+
+        this.setState({
+            roleOptions: [],
+            currentUserRoleId: '',
+            isAdmin: propsData.isAdmin,
+            imgWidth: propsData.imgWidth,
+            imgHeight: propsData.imgHeight,
+            loading: true,
+            closeDialog: false,
+            pageRenderedHtml: props.pageRenderedHtml
+        });
+
+        forkJoin([this.dotRolesService.search(), this.dotCurrentUser.getCurrentUser()])
+            .pipe(take(1))
+            .subscribe(([roles, currentUser]: [DotRole[], DotCurrentUser]): void => {
+                this.patchState({
+                    loading: false,
+                    currentUserRoleId: currentUser.roleId,
+                    roleOptions: roles
+                });
+            });
+    }
```

Comment:
If there are any visual change if you do this?


```suggestion
        forkJoin([this.dotRolesService.search(), this.dotCurrentUser.getCurrentUser()])
            .pipe(take(1))
            .subscribe(([roles, currentUser]: [DotRole[], DotCurrentUser]): void => {
                this.setState({
                    roleOptions: role,
                    currentUserRoleId: currentUser.roleId,
                    isAdmin: propsData.isAdmin,
                    imgWidth: propsData.imgWidth,
                    imgHeight: propsData.imgHeight,
                    loading: false,
                    closeDialog: false,
                    pageRenderedHtml: props.pageRenderedHtml
                });
            });
    }
```

Reply:
before the `forkJoin` we have not set the state at all, also before the `forkJoin` we need to set the `loading` as TRUE and it can not be done without initializing the state